pkg/sql/sem/tree/stmt.go

```diff
@@ -439,6 +439,17 @@ func (*AlterTenantSetClusterSetting) StatementType() StatementType { return Type
 // StatementTag returns a short string identifying the type of statement.
 func (*AlterTenantSetClusterSetting) StatementTag() string { return "ALTER TENANT SET CLUSTER SETTING" }
 
+// StatementReturnType implements the Statement interface.
+func (*AlterTenantReplication) StatementReturnType() StatementReturnType { return Rows }
+
+// StatementType implements the Statement interface.
+func (*AlterTenantReplication) StatementType() StatementType { return TypeDML }
+
+// StatementTag returns a short string identifying the type of statement.
+func (*AlterTenantReplication) StatementTag() string { return "ALTER TENANT REPLICATION" }
+
+func (*AlterTenantReplication) cclOnlyStatement() {}
```

Comment:
This is a general issue with a number of our Opaque statements at the moment, but I _think_ that this statement will still be allowed during read-only transactions.  One option would be to update CanWriteData in this same file.

Reply:
nice catch, added `CanWriteData` and a test.